compiler/src/yul/mappers/functions.rs

```diff
@@ -42,39 +42,42 @@ pub fn func_def(
             .collect::<Result<Vec<_>, _>>()?;
 
         // Different return types require slightly different functions.
-        if let Some(return_type) = attributes.return_type.to_owned() {
-            return match return_type {
-                // Base types are returned by value. All memory used by the function is cleared.
-                FixedSize::Base(_) => Ok(function_definition! {
+        return match attributes.return_type.to_owned() {
+            // Base types are returned by value. All memory used by the function is cleared.
+            FixedSize::Base(_) => Ok(function_definition! {
+                function [function_name]([param_names...]) -> return_val {
+                    (let ptr := avail())
+                    [function_statements...]
+                    (free(ptr))
+                }
+            }),
+            // Arrays need to keep memory allocated after completing.
+            // FIXME: Copy arrays to the lowest available pointer to save memory.
```

Comment:
We might as well delete this comment since we're on the fence about memory de-allocation.